### PR TITLE
Added Graph Feature Preprocessor Example 

### PR DESCRIPTION
**Pull request by andreea-anghel**

### CODE REVIEW

**Review by kubilay-atasu — COMMENTED** [review]
@andreea-anghel Even though no model training and inference is happening in this example, in real-life, one would perform the training also on the enriched feature vectors. Should we clarify that in this example by enriching also the train graph?

**Comment by kubilay-atasu** [comment]
In addition, it is not necessary to load the train graph before testing. Loading some past transactions, especially those that took place immediately before the test transactions would improve the quality of the features extracted for the test data, but it is not necessary to load the whole history. Maybe this also requires some clarification.

**Comment by kubilay-atasu** [comment]
So, I suggest that 1) we also do feature transformation on the training graph, 2) say that we can now train a model on the transformed training graph, 3) do feature transformation on the test graph, 4) say that we can now do inference on the transformed test graph using the model train. The input graphs are too small to train a meaningful model and test it accurately, but this is how the GraphFeaturePreprocessor would be used in real life. Do you agree?

**Comment by kubilay-atasu** [comment]
I do not think it is necessary to load an initial graph before training.  We would need to have some unlabelled initial data that precedes the labeled training data to justify doing that. I think it would be sufficient to show "enrich+train" followed by "enrich+test". We could also that fitting on the training data (or some historical data) before transforming the test data would improve the quality of the test features computed.

**Comment by andreea-anghel** [comment]
Ok, then we remove the `fit` from the example (we include a comment about it) and show only enrich+train/test.

**Comment by kubilay-atasu** [comment]
I see your point. We also need to show how and when `fit` should be used.

**Comment by kubilay-atasu** [comment]
So, then maybe we should indicate that this is only the inference code. Loading some historical data would help detect deeper features, but we probably should not call it the training data.

**Comment by kubilay-atasu** [comment]
Another possibility is to create two different GraphFeaturePreprocesor objects for training and testing, and have the test object call fit on the training data before performing inference on the test data. What do you think?

**Comment by andreea-anghel** [comment]
yes, I think I wanted to go in the same direction with the two use cases above.

*Prepare an ML model that incorporates the graph-related features*
    
    1. Create gfp1 
    2. Use gfp1 to enrich the transactions from the train set (labeled transactions, no previous fit)
    3. Use the transactions in 2 to train a ML model (as a comment, no actual model training)

*Run ML model inference to detect fraud on new transactions*

    1. Create gfp2 
    2. Load a graph thus run fit on gfp2 using the train data - we can say it is optional
    3. Enrich the test transactions using gfp2
    4. Run the ML model trained above on the new enriched transactions to get the predictions (as a comment, no actual inference)

File: examples/datasets/graph-feature-preprocessor/aml_custom_train.txt [hunk 1]
```diff
@@ -1,9 +1,13 @@
 # transaction ID, source account ID, target account ID, timestamp
 0 0 1 0
+0 0 1 0
+0 0 1 0
 1 1 2 1
 2 2 3 2
 3 1 3 3
 4 3 1 4
+4 3 1 4
+4 3 1 4
```
Comment (jovan-blanusa) [inline examples/datasets/graph-feature-preprocessor/aml_custom_train.txt:10]: FYI, this example works only with the new version of the graph preprocessor that I recently pushed (https://github.ibm.com/ai-foundation/snapml/pull/637). The previous version of the code would report an error due to duplicate edge IDs.

Comment (andreea-anghel) [reply examples/datasets/graph-feature-preprocessor/aml_custom_train.txt:10]: yes, thanks. I'm using the main version and it works.

**Review by kubilay-atasu — APPROVED** [review]
LGTM

**Pull request merged**